chainstate/src/detail/mod.rs

```diff
@@ -768,12 +770,11 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
 
         // TODO: Add a check for the chain trust.
 
-        let tip_timestamp = match self.query()?.get_best_block_header() {
-            Ok(h) => h.timestamp(),
-            // There is only the genesis block, so the initial block download isn't finished yet.
-            Err(PropertyQueryError::GenesisHeaderRequested) => return Ok(()),
-            Err(e) => return Err(e),
-        };
+        let tip_timestamp = self
+            .query()?
+            .get_best_block_index()?
+            .expect("Best block index should always exist")
+            .block_timestamp();
```

Comment:
What do you think about this? It changes genesis handling, so that it takes the timestamp into account rather than making genesis tip to be considered to always imply the initial block download state. This way, genesis and normal blocks are treated uniformly. However, there may be considerations for the different genesis handling that I had not realised.

Without this change, a bunch of mempool tests stops working because transactions get rejected due to IBD state (since the blockchain is at genesis initially).

Any comments @pavel-kokolemin @TheQuantumPhysicist ?

Reply:
I think this is a reasonable change.

Reply:
I'm OK with this.